### ISSUE TITLE
Cannot start container from Dockerfile

### ISSUE THREAD

**Issue by Vesli** [testcontainers/testcontainers-go#212]
**Describe the bug**
If I manually build and start my Dockerfile (docker build . -t test && docker run -t test) it works without problems.
But:
```
req := testcontainers.ContainerRequest{
		FromDockerfile: testcontainers.FromDockerfile{
			Context:    "../",
			Dockerfile: "Dockerfile",
		},
		ExposedPorts: []string{"8075/tcp"},
		Env:          env,
		Name:         "service",
		WaitingFor:   wait.ForListeningPort("8075/tcp"),
	}

serviceContainer, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
		ContainerRequest: req,
		Started:          true,
	})
```
returns: 
>/bin/sh command not executable

The image is built but refuses to run

Dockerfile:
```
FROM debian:bullseye-slim

RUN apt-get update && \
    apt-get install -y --no-install-recommends ca-certificates && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*

RUN groupadd -g 1000 gouser && \
    useradd -r -u 1000 -g gouser gouser
USER gouser

ARG CREATED=unspecified
ARG GIT_COMMIT=unspecified
ARG CIRCLE_REPOSITORY_URL=unspecified
ARG PROJECT=unspecified
ARG VERSION=unspecified
ARG CIRCLE_BUILD_URL=unspecified

LABEL org.opencontainers.image.authors=${AUTHOR} \
      org.opencontainers.image.created=${CREATED} \
      org.opencontainers.image.revision=${GIT_COMMIT} \
      org.opencontainers.image.source=${CIRCLE_REPOSITORY_URL} \
      org.opencontainers.image.title=${PROJECT} \
      org.opencontainers.image.version=${VERSION} \
      ai.adarga.circle-build-url=${CIRCLE_BUILD_URL}

############ end following ################

COPY bin/service bin/service

CMD ["./bin/service"]
```

Nonetheless, from the testcontainers image created, if I run:
>docker run -it <tag-generated>
or
>docker run -it <tag-generated> /bin/sh

both work without problem, the binary is at the right location within the container etc.

** docker info **
output of the command:

```
$ docker info
Client:
 Debug Mode: false

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 49
 Server Version: 19.03.8
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 1.944GiB
 Name: docker-desktop
 ID: RWR6:WDHQ:NHGL:KW5V:WAMR:5JBY:AFNW:ABK4:Z5NN:D5CF:IAGR:5GPS
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 40
  Goroutines: 52
  System Time: 2020-06-12T15:51:59.6351099Z
  EventsListeners: 3
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Additional context**
Sorry, pushing some issues as I really like the efforts made into this lib!

And here running the container should run without problems, right?

**Comment by Vesli** [comment]
On CircleCi the error is different, it cannot find the image it created from the Dockerfile:
>Error response from daemon: No such image: 9c532d58-b46d-4016-94f6-8bc6b1d22b1f:6c2a1f13-40f6-41ba-8b71-96767afe8e19

**Comment by ClaytonNorthey92** [comment]
hey @Vesli I apologize it took me so long to get back to you, could you do one thing for me?  ...

Could you try using a _full path_ rather than a relative path to your `Context`?  And could you tell me what error you see if you do that?

**Comment by Vesli** [comment]
Hello @ClaytonNorthey92 

So I tried and the same error happens:

>2020/06/16 10:30:13 Starting container id: 720fedc46f2a image: 3388f48c-6ffe-4972-9981-91917c66b421:0128955c-37bc-40be-acac-ddaa8a8fbb3b
2020/06/16 10:30:14 Container started!
2020/06/16 10:30:14 Waiting for container id 720fedc46f2a image: 3388f48c-6ffe-4972-9981-91917c66b421:0128955c-37bc-40be-acac-ddaa8a8fbb3b

And right away after "waiting for container" there is the error showing up:
>/bin/sh command not executable

I've tried with removing the env variables for my service, which should result with "cannot start with missing parameters" but only the /bin/sh error still shows

```
FromDockerfile: testcontainers.FromDockerfile{
			Context:    "/Users/jean-charlespassepont/Documents/path/my-service-root",
			Dockerfile: "Dockerfile",
		},
```

(I've tried with a ending `/` as well)

**Comment by gianarb** [comment]
It will help us a lot of you can strip out everything that is proprietary and you can serve to us an easy to reproduce project. It can be a tar archive loaded as an attachment to this issue

So we can reproduce it easily

**Comment by islishude** [comment]
It's not a bug,you should set the Dockerfile path with [os.Getwd](https://golang.google.cn/pkg/os/#Getwd)

**Comment by gianarb** [comment]
Gonna close this issue, for now, feel free to open another one if you need more information. We have to write documentation for this in order to make it clear.

**Comment by dtest** [comment]
I know this is an old issue, but I came across myself recently. Hopefully my findings help.

TL;DR: For the `wait.ForListeningPort("XXXX/tcp"),` command to work, the container must support net-tools (or whatever exposes [nc](https://github.com/testcontainers/testcontainers-go/blob/838c6e8ee90d3d0d4af5b733631613374c0d3d9c/wait/host_port.go#L126-128) for your image distro)

The problem is that my container was a very bare container (as most things should be, right?). Here's an example dockerfile that was broken:

```
FROM golang:alpine as builder
WORKDIR /app
COPY . .

RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o my-bin -ldflags="-w -s" .

FROM scratch

COPY --from=builder /app/my-bin /usr/bin/

EXPOSE 80

ENTRYPOINT ["my-bin"]
```

To get it working I switched from`scratch` to `alpine` and installed net-tools
```
FROM golang:alpine as builder
WORKDIR /app
COPY . .

RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o my-bin -ldflags="-w -s" .

FROM alpine:latest

RUN apk -U upgrade \
    && apk add --no-cache \
        net-tools \
    && rm -rf /var/cache/apk/*

COPY --from=builder /app/my-bin /usr/bin/

EXPOSE 80

ENTRYPOINT ["my-bin"]
```

Regarding doc updates, more examples on custom containers would be awesome, along with some debugging examples. I had to insert a Sleep to make sure the container was indeed starting up fine before digging in to the wait logic.

**Comment by mdelapenya** [comment]
Hey @dtest sorry of you found the lack of debugging examples disturbing :) If you are able to help us with the docs/examples, that would be awesome!!
